lib/matplotlib/backends/backend_agg.py

```diff
@@ -554,25 +565,29 @@ def print_jpg(self, filename_or_obj, *args, dryrun=False, **kwargs):
             color = tuple([int(x * 255) for x in rgba[:3]])
             background = Image.new('RGB', size, color)
             background.paste(image, image)
-            options = {k: kwargs[k]
-                       for k in ['quality', 'optimize', 'progressive', 'dpi']
-                       if k in kwargs}
-            options.setdefault('quality', rcParams['savefig.jpeg_quality'])
-            if 'dpi' in options:
-                # Set the same dpi in both x and y directions
-                options['dpi'] = (options['dpi'], options['dpi'])
-
-            return background.save(filename_or_obj, format='jpeg', **options)
+            if pil_kwargs is None:
+                pil_kwargs = {}
+            for k in ["quality", "optimize", "progressive"]:
```

Comment:
Are you specifically dropping `dpi` from here on purpose?

Reply:
print_figure() actually starts by applying the `dpi` kwarg with `self.figure.dpi = dpi` (resetting it to its original value at exit); see comment at top.
So the only case where this would matter is if one calls print_jpeg() directly instead of calling savefig() or even print_figure().  But in that case you're already not correctly applying the dpi anyways, so that's pretty close to unsupported IMO.
Also this behavior is consistent with print_tiff.